src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmExcelTemplateExportServiceImpl.java

```diff
@@ -183,6 +206,146 @@ private File generateTemplateFile(final String fileNamePath, final String cropNa
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_OLIVE_GREEN));
+		cell.setCellValue(this.getMessageSource().getMessage("export.germplasm.list.template.varible", null, locale));
+		sheet.setColumnWidth(0, (cell.getStringCellValue().length() + COLUMN_WIDTH_PADDING) * CHARACTER_WIDTH * 2);
+
+		cell = row.createCell(1, CellType.STRING);
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_OLIVE_GREEN));
+		cell.setCellValue(this.getMessageSource().getMessage("export.germplasm.list.template.alias", null, locale));
+		sheet.setColumnWidth(1, (cell.getStringCellValue().length() + COLUMN_WIDTH_PADDING) * CHARACTER_WIDTH * 2);
+
+		cell = row.createCell(2, CellType.STRING);
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_OLIVE_GREEN));
+		cell.setCellValue(this.getMessageSource().getMessage("export.germplasm.list.template.description", null, locale));
+		sheet.setColumnWidth(2, (cell.getStringCellValue().length() + COLUMN_WIDTH_PADDING) * CHARACTER_WIDTH * 2);
+
+		cell = row.createCell(3, CellType.STRING);
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_OLIVE_GREEN));
+		cell.setCellValue(this.getMessageSource().getMessage("export.germplasm.list.template.porperty", null, locale));
+		sheet.setColumnWidth(3, (cell.getStringCellValue().length() + COLUMN_WIDTH_PADDING) * CHARACTER_WIDTH * 2);
+
+		cell = row.createCell(4, CellType.STRING);
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_OLIVE_GREEN));
+		cell.setCellValue(this.getMessageSource().getMessage("export.germplasm.list.template.scale", null, locale));
+		sheet.setColumnWidth(4, (cell.getStringCellValue().length() + COLUMN_WIDTH_PADDING) * CHARACTER_WIDTH * 2);
+
+		cell = row.createCell(5, CellType.STRING);
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_OLIVE_GREEN));
+		cell.setCellValue(this.getMessageSource().getMessage("export.germplasm.list.template.method", null, locale));
+		sheet.setColumnWidth(5, (cell.getStringCellValue().length() + COLUMN_WIDTH_PADDING) * CHARACTER_WIDTH * 2);
+
+		cell = row.createCell(6, CellType.STRING);
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_OLIVE_GREEN));
+		cell.setCellValue(this.getMessageSource().getMessage("export.germplasm.list.template.data.type", null, locale));
+		sheet.setColumnWidth(6, (cell.getStringCellValue().length() + COLUMN_WIDTH_PADDING) * CHARACTER_WIDTH * 2);
+
+		cell = row.createCell(7, CellType.STRING);
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_OLIVE_GREEN));
+		cell.setCellValue(this.getMessageSource().getMessage("export.germplasm.list.template.expected.values", null, locale));
+		sheet.setColumnWidth(7, (cell.getStringCellValue().length() + COLUMN_WIDTH_PADDING) * CHARACTER_WIDTH * 2);
+
+		return rowNumIndex;
+	}
+
+	private String getExpectedRange(final String dataTypeName, final VariableDetails variableDetails) {
```

Comment:
@cuenyad 
I think it will be more elegant to receive DataType class

Reply:
We should move this block of code to an Util class so it can be reused when exporting the fieldbook in Manage Studies. I will confirm if we need it for the Germplasm lists